### PR TITLE
CircleCI: Lint code for undefined names

### PR DESCRIPTION
**Pull request by cclauss**
Add [flake8](http://flake8.pycqa.org) tests to find Python syntax errors and undefined names.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-torch==1.0.1.post2
+torch>=1.1.0
```
Comment (apsdehal) [inline requirements.txt]: We are not yet ready to make this migration to 1.1.0. Can you revert this? It causes some issues in data parallel for us, so we need to figure it out properly before making this change.

Comment (cclauss) [reply requirements.txt]: Reverted but that pegging that torch version causes the build to fail.

**Comment by apsdehal** [comment]
Let's fix `torchvision==0.2.2`. This should fix the build.

**Comment by apsdehal** [comment]
Let me merge this and fix the build issue on my end. Thanks.

**Comment by bachagakshokh** [comment]
Yusaf Adfgh

**Pull request merged**